### ISSUE TITLE
Question on Ensemble mode 

### ISSUE THREAD

**Issue by leepeter2019** [trailofbits/deepstate#275]
I have question on ensemble mode
When i test the ensemble mode with AFL and Honggfuzz, Deepstate can copy files in local queue to global sync folder and vice versa as you implemented.
But each Fuzzer (AFL, Honggfuzz) doesnot add new files in queue folder into their own queue.
the number of total paths does not changed at all. the result also showed that they didnot use new seed file.
Have you ever test ensemble mode and check they use new seed files?

**Comment by ex0dus-0x** [comment]
Thanks for bringing this to concern. One idea for why such a behavior is being exhibited is the fact that fuzzers may internally distillate corpuses in order to prioritize ones that contribute to higher coverage. As mentioned in the AFL [internal specification](http://lcamtuf.coredump.cx/afl/README.txt):

```
The discovered test cases are also periodically culled to eliminate ones that
have been obsoleted by newer, higher-coverage finds; and undergo several other
instrumentation-driven effort minimization steps.
```

While this may not necessarily be a case, meaning if we do have fuzzers that are actively generating interesting and diverse seeds individually, a proposed method to better guide our seed synchronization might be implementing an "oracle" of some sort that either internally ranks seed diversity (hence a "smarter" ensemble-fuzzer),  and/or one that stops/resumes fuzzers in an effort to "re-seed" from their respective queues.

**Comment by GrosQuildu** [comment]
Synchronization is somehow fixed by #326 . But comment above is correct - some fuzzers (i.e. Hongfuzz) need restart to sync.
 
See #331 for more informations.

Closing it.
